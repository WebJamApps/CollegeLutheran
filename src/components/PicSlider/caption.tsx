import React from 'react';

const Caption = (props: { caption?: (JSX.Element|string); }):JSX.Element => {
  const { caption } = props;
  return (
    <div
      className="slider-caption"
      style={{
        textAlign: 'center',
        fontWeight: 'bold',
        padding: 0,
        backgroundColor: 'silver',
        marginBottom: 0,
        marginTop: 0,
      }}
    >
      {' '}
      {caption}
      {' '}
    </div>
  );
};

Caption.defaultProps = {
  caption: '',
};

export default Caption;
