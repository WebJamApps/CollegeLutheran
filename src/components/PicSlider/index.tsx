import React, { Component } from 'react';
import Slider from 'react-slick';
import 'slick-carousel/slick/slick.css';
import 'slick-carousel/slick/slick-theme.css';
import Caption from './caption';
import { Ibook } from '../../redux/mapStoreToProps';

type SettingsType = {
  autoplay: boolean;
  autoplaySpeed: number;
  infinite: boolean;
  speed: number;
  slidesToShow: number;
  slidesToScroll: number;
  arrows: boolean;
  fade: boolean;
};

export interface PicSliderProps {
  settings?: SettingsType;
  data: Ibook[];
}
class PicSlider extends Component<PicSliderProps> {
  static defaultProps: { data: [{ url: ''; title: ''; _id: 0 }]; };

  settings: SettingsType;

  constructor(props: Readonly<PicSliderProps>) {
    super(props);
    this.settings = {
      autoplay: true,
      autoplaySpeed: 3000,
      infinite: true,
      speed: 5000,
      slidesToShow: 1,
      slidesToScroll: 1,
      arrows: false,
      fade: true,
    };
  }

  /* eslint-disable react/jsx-props-no-spreading */
  render(): JSX.Element {
    const { data } = this.props;
    return (
      <div>
        <Slider {...this.settings}>
          {
            data.map((d: Ibook) => (
              <div key={d._id}>
                <img className="slide-show" src={d.url} alt={d.title} />
                {d.comments === 'showCaption' ? <Caption caption={d.title} /> : null}
              </div>
            ))
          }
        </Slider>
      </div>
    );
  }
}

export default PicSlider;
