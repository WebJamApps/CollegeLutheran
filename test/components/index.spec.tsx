import renderer from 'react-test-renderer';
import PicSlider from 'src/components/PicSlider';

describe('PicSlider', () => {
  it('renders with a caption', () => {
    const data = [
      {
        type: 'otherPics', url: 'https://test.com', title: 'test', _id: '123', comments: '',
      },
      {
        type: 'otherPics', comments: 'showCaption', url: 'https://test2.com', title: 'test2', _id: '345',
      },
    ];
    const result: any = renderer.create(<PicSlider data={data} />).toJSON();
    expect(result).toMatchSnapshot();
  });
});
