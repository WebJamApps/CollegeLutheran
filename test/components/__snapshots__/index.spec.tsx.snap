// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PicSlider renders with a caption 1`] = `
<div
  id="pic-slider"
>
  <div
    className="slick-slider slick-initialized"
    dir="ltr"
  >
    <div
      className="slick-list"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={null}
      onMouseMove={null}
      onMouseUp={[Function]}
      onTouchCancel={null}
      onTouchEnd={[Function]}
      onTouchMove={null}
      onTouchStart={[Function]}
      style={{}}
    >
      <div
        className="slick-track"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <div
          aria-hidden={false}
          className="slick-slide slick-active slick-current"
          data-index={0}
          onClick={[Function]}
          style={
            {
              "left": -0,
              "opacity": 1,
              "outline": "none",
              "position": "relative",
              "transition": "opacity 5000ms ease, visibility 5000ms ease",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <div
              style={
                {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            >
              <img
                alt="test"
                className="slide-show"
                src="https://test.com"
              />
            </div>
          </div>
        </div>
        <div
          aria-hidden={true}
          className="slick-slide"
          data-index={1}
          onClick={[Function]}
          style={
            {
              "left": -0,
              "opacity": 0,
              "outline": "none",
              "position": "relative",
              "transition": "opacity 5000ms ease, visibility 5000ms ease",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <div
              style={
                {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            >
              <img
                alt="test2"
                className="slide-show"
                src="https://test2.com"
              />
              <div
                className="slider-caption"
                style={
                  {
                    "backgroundColor": "silver",
                    "fontWeight": "bold",
                    "marginBottom": 0,
                    "marginTop": 0,
                    "padding": 0,
                    "textAlign": "center",
                  }
                }
              >
                 
                test2
                 
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
